### PR TITLE
Adds non-verified option to the payment page

### LINKED ISSUE #8231: Add ledger switch to disable contributions to non-verified sites (in advanced settings)
image

If the switch is disabled, the ledger should only display verified sites. Additionally, the "contribute" button on the right side of the URL bar should only appear for sites that are verified.

Text string for switch: "Allow contributions to non-verified sites"

cc @mrose17

### PR DESCRIPTION
**Pull request by NejcZdovc**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Resolves #8231

Auditors: @mrose17 @bsclifton

### Test Plan
- go to payments page and enable payments
- go to some verified and non-verified sites (brianbondy.com is verified)
- go to payments page
- click on advance settings
- disabled option "Allow contributions to non-verified sites"
- all non-verified sites should be hidden from a table

### CODE REVIEW

**Comment by NejcZdovc** [comment]
@mrose17, these PR is not yet completed, still have work-in-progress label

File: js/constants/appConfig.js [hunk 1]
```diff
@@ -149,6 +149,7 @@ module.exports = {
     'bookmarks.toolbar.showOnlyFavicon': false,
     'payments.enabled': false,
     'payments.notifications': false,
+    'payments.non-verified-publishers': true,
```
Comment (bsclifton) [inline js/constants/appConfig.js]: Could you update this to be more descriptive? Maybe `payments.allow-non-verified-publishers`?

File: js/constants/settings.js [hunk 2]
```diff
@@ -55,6 +55,7 @@ const settings = {
   PAYMENTS_NOTIFICATION_RECONCILE_SOON_TIMESTAMP: 'notification-reconcile-soon-timestamp',
   PAYMENTS_NOTIFICATION_TRY_PAYMENTS_DISMISSED: 'payments.notificationTryPaymentsDismissed',
   PAYMENTS_CONTRIBUTION_AMOUNT: 'payments.contribution-amount',
+  PAYMENTS_NON_VERIFIED: 'payments.non-verified-publishers',
```
Comment (bsclifton) [inline js/constants/settings.js]: same as above comment 😄

File: js/constants/appConfig.js (hunk 1 above)
Comment (NejcZdovc) [inline js/constants/appConfig.js]: done

File: js/constants/settings.js (hunk 2 above)
Comment (NejcZdovc) [inline js/constants/settings.js]: done

**Pull request merged**